File: api-asn/csharp/src/api/asn/x509/EVersion.cs
using Version = tr.gov.tubitak.uekae.esya.asn.x509.Version;
namespace tr.gov.tubitak.uekae.esya.api.asn.x509
{
    public class EVersion : BaseASNWrapper<Version>
    {
        public static readonly EVersion v1 = new EVersion(Version.v1);
        public static readonly EVersion v2 = new EVersion(Version.v2);
        public static readonly EVersion v3 = new EVersion(Version.v3);

        public EVersion(byte[] aBytes)
            : base(aBytes, new Version())
        {
        }

        public EVersion(long version)
            : base(new Version(version))
        {
        }
        public long getVersion()
        {
            return mObject.mValue;
        }
    }
}

File: api-asn/src/tr/gov/tubitak/uekae/esya/api/asn/x509/EVersion.java
package tr.gov.tubitak.uekae.esya.api.asn.x509;

import tr.gov.tubitak.uekae.esya.api.asn.BaseASNWrapper;
import tr.gov.tubitak.uekae.esya.api.common.ESYAException;
import tr.gov.tubitak.uekae.esya.asn.x509.Version;

/**
 * User: zeldal.ozdemir
 * Date: 1/28/11
 * Time: 10:20 AM
 */
public class EVersion extends BaseASNWrapper<Version>{
   public final static EVersion v1 = new EVersion(Version.v1);
   public final static EVersion v2 = new EVersion(Version.v2);
   public final static EVersion v3 = new EVersion(Version.v3);

    public EVersion(byte[] aBytes) throws ESYAException {
        super(aBytes, new Version());
    }

    public EVersion(long version) {
        super(new Version(version));
    }

    public long getVersion(){
        return mObject.value;
    }
}
